### PR TITLE
feat: add index.d.ts

### PR DESCRIPTION
**Pull request by aleclarson**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {Plugin} from 'unified'
+
+declare const remarkSlug: Plugin
```
Comment (ChristianMurphy) [inline index.d.ts:3]: the default generic on plugin accepts a single object like setting.
This plugin does not accept any settings, which can be specified with `<[]>`

```suggestion
declare const remarkSlug: Plugin<[]>
```

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]
Thanks @aleclarson!
Could we add dtslint and a type test, similar to https://github.com/remarkjs/remark-breaks/pull/1 ?

**Comment by aleclarson** [comment]
> Could we add dtslint and a type test, similar to [remarkjs/remark-breaks#1](https://github.com/remarkjs/remark-breaks/pull/1) ?

Seems like overkill for what's basically a one liner?

**Comment by ChristianMurphy** [comment]
There are 300+ repos across unified, syntax tree, remark, rehype, retext, and redot; with types being added to each over time.
Ensuring a basic level of quality for all the types is a goal.
Understanding what versions of TS are supported, and having a test in place for options or in case options are added, is valuable at an ecosystem level.

So to answer your question, no, it's isn't overkill.
The context is broader than a single repo with several line typing.

**Comment by aleclarson** [comment]
If you're worried about that, maybe you could write a script I can run to set that up? Otherwise, these types work fine as of now. Someone else can contribute the test setup if they find it necessary.

**Comment by ChristianMurphy** [comment]
> you could write a script I can run to set that up?

the setup is
```
npm install --save-dev dtslint
```

the tsconfig and tslint files shift from project to project, but you could use the ones from remarkjs/remark-breaks#1 as a starting point.
If you want to automate that more, feel free.

> Otherwise, these types work fine as of now

I'm sure it does for fine for you.
And with some test tooling in place, it can also match the community standards and be maintainable long term.

> Someone else can contribute the test setup if they find it necessary.

Or you could add it.
I'm personally not interested in merging an incomplete PR.
The PR is close, just a bit more and you can finish what you started.

**Comment by wooorm** [comment]
New features should come with tests: https://github.com/remarkjs/.github/blob/main/contributing.md#submitting-a-pull-request.

Just a type file doesn't do anything, and this is failing existing tests. I don't think this is a weird or ambitious ask.

**Comment by aleclarson** [comment]
Okay, should we keep this open until someone with the necessary willpower forks it and opens a follow-up PR? In the meantime, I've published [`@alloc/remark-slug`](https://www.npmjs.com/package/@alloc/remark-slug) with the types.

**Comment by remcohaszing** [comment]
@unifiedjs / @remarkjs packages merely upholds the same quality standards for types as followed by [DefinitelyTyped/DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). I know this is a very welcoming community willing to help first time contributors, especially @ChristianMurphy in my experience.

When contributing to an open source project, one should adhere to the quality standards of that project, even if they don’t like it. This doesn’t only apply to @unifiedjs, but to **any** open source contributions.

- Always read the contributing guidelines.
- Always adhere to the coding standards.
- Does a project have tests? Write tests for your contribution.
- Does a project certain conventions? Adhere to those conventions.
- Do you get feedback on your contribution? Process that feedback.

Not only is creating and maintaining a fork more work, also this contributes to fragmentation and unclarity within the JavaScript community. I hope you decide to deprecate `@alloc/remark-slug`. Otherwise, please fix your project metadata.

- Your default branch is behind, `master` is published instead.
- The project metadata should link to your fork, not to [remarkjs/remark-slug](https://github.com/remarkjs/remark-slug).
- The readme should state why the project was forked.

**Comment by wooorm** [comment]
Going to close this! I am a bit flabbergasted by wanting to land a broken feature without tests, but if the author is not interested in working on this anymore, that's totally okay

**Comment by aleclarson** [comment]
> * Your default branch is behind, `master` is published instead.
> * The project metadata should link to your fork, not to [remarkjs/remark-slug](https://github.com/remarkjs/remark-slug).
> * The readme should state why the project was forked.

All fixed :)



> wanting to land a broken feature

It's working fine for me :)

**Comment by wooorm** [comment]
> It's working fine for me :)

For one, types aren’t published, although I saw you’ve added that to your fork.

> All fixed :)

👍

**Comment by SalahAdDin** [comment]
@aleclarson what happened with this?

**Comment by ChristianMurphy** [comment]
@SalahAdDin you can see what happened in the thread above you comment :slightly_smiling_face: 
It remains unmerged because it doesn't include any type tests.

**Comment by ChristianMurphy** [comment]
For folks coming across this thread in the future, @SalahAdDin was kind enough to open a follow up PR with types and type tests https://github.com/remarkjs/remark-slug/pull/19
The types are now included in version https://github.com/remarkjs/remark-slug/releases/tag/6.1.0 and above.

**Pull request closed without merging**